File: Prácticas/P3/Ruby/civitas/lib/juego_texto.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative "vista_textual.rb"
require_relative "controlador.rb"
require_relative "dado.rb"
require_relative "civitas_juego.rb"

module JuegoTexto
  class JuegoTexto
    include Civitas
    def self.main
      @vista = Vista_textual.new
      
      @nombres = Array.new()
      @nombres << "Jose"
      @nombres << "Alex"
      @nombres << "Ruben"
      
      @@dado = Dado.instance 
      @@dado.debug = true
      
      @juego = CivitasJuego.new(@nombres)
      @controller = Controlador.new(@juego,@vista)
      @controller.juega()
    end
  end
  JuegoTexto.main
end

File: Prácticas/P3/Java/civitas/src/JuegoTexto/JuegoTexto.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package JuegoTexto;

/**
 *
 * @author Jose Padial
 */

import civitas.CivitasJuego;
import civitas.Dado;
import java.util.ArrayList;

public class JuegoTexto {

    /**
     * @param args the command line arguments
     */
    public static void main(String[] args) {
        VistaTextual vista= new VistaTextual();
        ArrayList<String> nombres = new ArrayList<>();
        nombres.add("Jose");
        nombres.add("Alex");
        nombres.add("Ruben");
        Dado.getInstance().setDebug(true);
        CivitasJuego juego= new CivitasJuego(nombres); 
        Controlador controller= new Controlador(juego, vista);
        controller.juega();
    }
    
}